compiler/src/test/java/com/squareup/anvil/compiler/codegen/BindingModuleQualifierTest.kt

```diff
@@ -3,51 +3,30 @@ package com.squareup.anvil.compiler.codegen
 import com.google.common.truth.Truth.assertThat
 import com.squareup.anvil.annotations.MergeComponent
 import com.squareup.anvil.annotations.MergeSubcomponent
-import com.squareup.anvil.annotations.compat.MergeModules
+import com.squareup.anvil.annotations.compat.MergeInterfaces
 import com.squareup.anvil.compiler.anyQualifier
-import com.squareup.anvil.compiler.compile
 import com.squareup.anvil.compiler.componentInterface
 import com.squareup.anvil.compiler.contributingInterface
 import com.squareup.anvil.compiler.generatedBindingModule
 import com.squareup.anvil.compiler.generatedMultiBindingModule
 import com.squareup.anvil.compiler.internal.testing.isAbstract
-import com.squareup.anvil.compiler.isFullTestRun
 import com.squareup.anvil.compiler.mergedModules
 import com.squareup.anvil.compiler.parentInterface
 import com.squareup.anvil.compiler.subcomponentInterface
+import com.squareup.anvil.compiler.testing.AnvilAnnotationsTest
 import dagger.Binds
 import dagger.Provides
 import dagger.multibindings.IntoSet
-import org.junit.Test
-import org.junit.runner.RunWith
-import org.junit.runners.Parameterized
-import org.junit.runners.Parameterized.Parameters
+import org.junit.jupiter.api.TestFactory
 import javax.inject.Named
-import kotlin.reflect.KClass
 
-@RunWith(Parameterized::class)
-class BindingModuleQualifierTest(
-  private val annotationClass: KClass<out Annotation>,
+class BindingModuleQualifierTest : AnvilAnnotationsTest(
+  MergeComponent::class,
+  MergeSubcomponent::class,
+  MergeInterfaces::class,
 ) {
 
-  private val annotation = "@${annotationClass.simpleName}"
-  private val import = "import ${annotationClass.java.canonicalName}"
-
-  companion object {
-    @Parameters(name = "{0}")
-    @JvmStatic
-    fun annotationClasses(): Collection<Any> {
-      return buildList {
-        add(MergeComponent::class)
-        if (isFullTestRun()) {
-          add(MergeSubcomponent::class)
-          add(MergeModules::class)
-        }
-      }
-    }
-  }
```

Comment:
Is the move to kase necessary here? I feel like the further these details move away from the test, the harder it is to reason about and maintain vs keeping it simple and using the Junit apis directly. E.g. it's no longer clear that this test suite is impacted by the `isFullTestRun()` check

Reply:
It's not *strictly* necessary.  Most of these tests were failing at some point during or after `ignoreQualifier` change, and it's much easier to debug them when you have the consistent working directories from Kase.  Any time I run into a similar situation with JUnit 4 tests, I've been migrating them over to Kase.

I would also argue that the mechanism for parameterized tests in JUnit 4 is more magical than simple -- at least for me.